alephium/test/token-bridge-test.ts

```diff
@@ -0,0 +1,610 @@
+        expect(events[0].fields).toEqual({
+            'sender': tokenBridgeInfo.contractId,
+            'sequence': 0,
+            'nonce': nonceHex,
+            'payload': toHex(message.encode()),
+            'consistencyLevel': 0
+        })
+    })
+
+    it('should register chain', async () => {
+        const tokenBridgeInfo = await createTokenBridge(provider)
+        const tokenBridge = tokenBridgeInfo.contract
+        const remoteTokenBridgeId = toHex(randomBytes(32))
+        const remoteChainId = CHAIN_ID_ALEPHIUM + 1
+        const registerChain = new RegisterChain(CHAIN_ID_ALEPHIUM, remoteChainId, remoteTokenBridgeId)
+        const vaaBody = new VAABody(registerChain.encode(), governanceChainId, governanceEmitterAddress, 0)
+        const vaa = initGuardianSet.sign(initGuardianSet.quorumSize(), vaaBody)
+        const testResult = await tokenBridge.testPublicMethod(provider, 'registerChain', {
+            address: tokenBridgeInfo.address,
+            initialFields: tokenBridgeInfo.selfState.fields,
+            testArgs: {
+                'vaa': toHex(vaa.encode()),
+                'payer': payer,
+                'createContractAlphAmount': minimalAlphInContract
+            },
+            inputAssets: [inputAsset],
+            existingContracts: tokenBridgeInfo.dependencies
+        })
+
+        const output = testResult.txOutputs[0]
+        const path = Buffer.concat([
+            chainIdToBytes(remoteChainId),
+            Buffer.from(tokenBridgeInfo.contractId, 'hex')
+        ])
+        const expectedContractId = Buffer.from(doubleHash(path)).toString('hex')
+        expect(toContractId(output.address)).toEqual(expectedContractId)
+        expect(BigInt(output.alphAmount)).toEqual(minimalAlphInContract)
+    })
+
+    it('should register chain failed if sequence is invalid', async () => {
+        const tokenBridgeInfo = await createTokenBridge(provider)
+        const tokenBridge = tokenBridgeInfo.contract
+        const registerChain = new RegisterChain(CHAIN_ID_ALEPHIUM, CHAIN_ID_ALEPHIUM + 1, randomContractId())
+        const vaaBody = new VAABody(registerChain.encode(), governanceChainId, governanceEmitterAddress, 1)
+        const vaa = initGuardianSet.sign(initGuardianSet.quorumSize(), vaaBody)
+        await expectAssertionFailed(async () => {
+            await tokenBridge.testPublicMethod(provider, 'registerChain', {
+                address: tokenBridgeInfo.address,
+                initialFields: tokenBridgeInfo.selfState.fields,
+                testArgs: {
+                    'vaa': toHex(vaa.encode()),
+                    'payer': payer,
+                    'createContractAlphAmount': minimalAlphInContract
+                },
+                inputAssets: [inputAsset],
+                existingContracts: tokenBridgeInfo.dependencies
+            })
+        })
+    })
+
+    it('should create token wrapper for local token', async () => {
```

Comment:
Is it possible to test that even if the sequence is out of order, it can still be processed as long as its not processed before? (If I understand correctly)

Reply:
The test case is here: https://github.com/alephium/wormhole-fork/pull/5/files#diff-48dda3fbdbf4f3b4cdc3e31e6c1b1d3583b5e9dbaf658415b81c4f57db974263R134, I will add more sequence related test cases to the token bridge.

Reply:
Awesome, thanks!